presto-raptor/src/main/java/com/facebook/presto/raptor/RaptorMetadata.java

```diff
@@ -780,6 +790,47 @@ private boolean viewExists(ConnectorSession session, SchemaTableName viewName)
         return !getViews(session, viewName.toSchemaTablePrefix()).isEmpty();
     }
 
+    @Override
+    public RaptorTableIdentity getTableIdentity(ConnectorTableHandle connectorTableHandle)
+    {
+        requireNonNull(connectorTableHandle, "connectorTableHandle is null");
+        RaptorTableHandle handle = checkType(connectorTableHandle, RaptorTableHandle.class, "connectorTableHandle");
+        return new RaptorTableIdentity(handle.getTableId());
+    }
+
+    @Override
+    public Map<String, ColumnIdentity> getColumnIdentities(ConnectorTableHandle connectorTableHandle)
+    {
+        requireNonNull(connectorTableHandle, "connectorTableHandle is null");
+        RaptorTableHandle handle = checkType(connectorTableHandle, RaptorTableHandle.class, "connectorTableHandle");
+        return getColumnHandles(handle).entrySet().stream()
+                .collect(toMap(entry -> entry.getKey(), entry -> new RaptorColumnIdentity(((RaptorColumnHandle) entry.getValue()).getColumnId())));
+    }
+
+    @Override
+    public RaptorTableIdentity deserializeTableIdentity(byte[] bytes)
+    {
+        requireNonNull(bytes, "bytes for TableIdentity is null");
+        if (bytes.length != Long.BYTES) {
```

Comment:
I would move the deserialize implementation to either a constructor on the identity objects or a static factory method on the identity object, so the reader can see the serialize and deserialize code in one file.